### PR TITLE
Make ConvertToVersion maybe 1090% faster on average

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

# Notes

* [Stats](https://docs.google.com/spreadsheets/d/1WJYd9QdDFTtaEQmtBzF0KWjfvUW7nVc2_nPekiE-6uM/edit#gid=191966452)

### CODE REVIEW

File: util/kube/convert.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+
+	"github.com/argoproj/argo-cd/util/tracing"
+)
+
+func convertToVersionWithScheme(obj *unstructured.Unstructured, group string, version string) (*unstructured.Unstructured, error) {
+	span := tracing.StartSpan("convertToVersionWithScheme")
+	defer span.Finish()
+	s := legacyscheme.Scheme
+	target := schema.GroupVersionKind{
```
Comment (alexec) [inline util/kube/convert.go]: simplify?

File: util/kube/ctl.go [hunk 2]
```diff
@@ -359,7 +359,19 @@ func (k *KubectlCmd) ConvertToVersion(obj *unstructured.Unstructured, group stri
 	if from.Group == group && from.Version == version {
 		return obj.DeepCopy(), nil
 	}
+	out, err := convertToVersionWithScheme(obj, group, version)
+	if err != nil {
```
Comment (alexec) [inline util/kube/ctl.go]: fix

**Review by jessesuen — COMMENTED** [review]
One simple optimization which we never did, but desperately need to, is to completely avoid attempting conversion of CR instances. These are a huge source of unnecessary work that is happening during reconciliation.  Can we make that part of this fix?

Second, can we add some tests for conversions which we know will fail?

**Review by alexmt — APPROVED** [review]
LGTM. @jessesuen

**Pull request merged**